params/config.go

```diff
@@ -204,6 +249,9 @@ type ChainConfig struct {
 	// Various consensus engines
 	Ethash *EthashConfig `json:"ethash,omitempty"`
 	Clique *CliqueConfig `json:"clique,omitempty"`
+
+	// Checkpoint contract configs
+	CheckpointContract *CheckpointContractConfig `json:"checkpointContract,omitempty"`
```

Comment:
I think I'd rename this into `CheckpointConfig`. Otherwise the `checkpointContract` name would suggest to me that it's an address, not an extended list of key-values.